### PR TITLE
Major frontend refactoring, started implementing the modal component.

### PR DESCRIPTION
**Pull request by dreamsmasher**

### CODE REVIEW

File: client/renderHelpers.js [hunk 1]
```diff
@@ -73,14 +77,7 @@ let cast = (konstruktor, value) => do {
     // eslint-disable-next-line @babel/new-cap
```
Comment (herbalmanatee) [inline client/renderHelpers.js:77]: !!

Comment (dreamsmasher) [reply client/renderHelpers.js:77]: ??

File: client/components/StatsTable.jsx [hunk 2]
```diff
@@ -3,61 +3,81 @@ import PropTypes from 'prop-types';
 import RatingsBar from './RatingsBar.jsx';
 import {aggregate} from '../renderHelpers.js';
 
+// exported in case we need more flexibility
```
Comment (herbalmanatee) [inline client/components/StatsTable.jsx:6]: nice use of react fragments

File: client/style/reviewBar.scss [hunk 3]
```diff
@@ -150,8 +150,38 @@ body {
   cursor: pointer;
 }
 
+@mixin bicolumn {
+  width: calc(50% - 0.5rem);
+}
+
```
Comment (herbalmanatee) [inline client/style/reviewBar.scss:171]: nesting!!

File: client/components/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React, {useState} from 'react';
```
Comment (herbalmanatee) [inline client/components/Modal.jsx:1]: I like how you destructure the props, gonna have to adopt this

**Review by herbalmanatee — APPROVED** [review]
Looks great. If you haven't already I would suggest looking into styled components. It helps clean up all the className's from your components and you can still nest styles within broader ones like in scss

File: client/components/Modal.jsx (hunk 4 above)
Comment (dreamsmasher) [reply client/components/Modal.jsx:1]: Haha this is just the conventional way to import functions defined outside of `export default`. Really useful.

**Pull request merged**